### DISCUSSION TITLE
Smart Meter Interface: Epever MPPT Tracer AN Serie 10-40A via Modbus

### DISCUSSION THREAD

**Discussion by luckysession** [arendst/Tasmota#12800] [Show and tell]
Hi guys,

my solution for data-logging via modbus, maybe someone need ... or add as an example @ [Smart Meter Descriptor examples](https://tasmota.github.io/docs/Smart-Meter-Interface/#smart-meter-descriptor-examples)

RS485 Adapter 485 Serieller Port UART Level Converter via Serial2 @ESP32

as scipting:
```
>D
;
>B
smlj=0
;->sensor53 r
>R
smlj=0
;
>S
if upsecs>30
then
smlj=1
endif
;
>M 1
;
+1,16,m,16,9600,,17,10,r010431100001,r0104311A0001,r0104331A0001,r0104331B0001,r010431000001,r010431010001,r010431020001,r0104310C0001,r0104310D0001,r0104310E0001,r010433040001,r0104330A0001,r0104330C0001,r010433120001,r010433030001
;
1,=h Batterie
1,010402SSssxxxx@i1:100,Temp,°C,Temperature,2
1,010402SSssxxxx@i2:1,SOC,%,Humidity,0
1,010402SSssxxxx@i3:100,Spannung,V,Voltage,2
1,010402SSssxxxx@i4:100,Strom,A,Current,2
1,=h PV-Modul
1,010402SSssxxxx@i5:100,Spannung,V,Voltage,2
1,010402SSssxxxx@i6:100,Strom,A,PV Current,2
1,010402SSssxxxx@i7:100,Power,W,PV Power,2
1,=h LOAD
1,010402SSssxxxx@i8:100,Spannung,V,Voltage,2
1,010402SSssxxxx@i9:100,Strom,A,Current Load,2
1,010402SSssxxxx@i10:100,Power,W,Power,2
1,=h Energie
1,010402SSssxxxx@i11:100,Verbrauch_daily,KWh,Today,2
1,010402SSssxxxx@i12:100,Verbrauch_total,KWh,Total out,2
1,010402SSssxxxx@i13:100,Ertrag_daily,KWh,Today in,2
1,010402SSssxxxx@i14:100,Ertrag_total,KWh,Total,2
1,010402SSssxxxx@i15:100,max.Batt.,V,Voltage_0,2
;
#`
```
![Unbenannt](https://user-images.githubusercontent.com/57873076/127740947-43d17081-c079-45dd-b1c0-d64616290d6e.PNG)

**Comment by Jaaxy** [comment]
i´m very interested in this issue, because i have an autark photovoltaik epever in range auf my WLAN.

**Reply by luckysession** [reply]
Try the script above!

**Reply by Jaaxy** [reply]
Ich habe das Script ausprobiert, allerdings noch nicht zum laufen gebracht. Mit dem USB-Adapter und der EPEver Software habe ich die Verbindung:
![image](https://user-images.githubusercontent.com/24737135/148093542-173eb0cf-e676-45fa-8313-01cd1cac2849.png)
Bei der Software musste ich 115200,8,N,1 einstellen, was mit dem Script schon mal nicht übereinstimmt.

Am ESP habe ich GPIO1 - TX mit RS485 TX und GPIO3 RX mit RS485 RX verbunden.
Eine Verbindung bekomme ich nicht. Als Adapter habe ich denselben wie für den SDM530 verwendet.

![image](https://user-images.githubusercontent.com/24737135/148094582-2683af58-f6ee-499f-9f10-94214fec58c7.png)

**Comment by luckysession** [comment]
Natürlich stimmen die Einstellungen nicht im Script, ich habe ein ESP32 und davon TX2 und RX2 genutzt.
Die sind über den UART-Level-converter angeschlossen, bei der Baudrate von 115200 läuft es bei mir am stabilsten... natürlich muss der epever auch gleich eingestellt sein, ist aber glaube ich default...
Leider weiß ich aber nicht ob RX auf TX zwischen dem Levelscoverter und dem ESP oder RX-RX, bzw. TX-TX, das entzieht sich gerade meiner Erinnerung und richtig dokumentiert habe ich das leider auch nicht.
Ich würde das gerne sofort nachgucken, aber leider voraussichtlich nicht vor dem WE, sonst ist es draußen zu dunkel... und außerdem fürchterlich am regnen.

Habe das hier noch gefunden... https://github.com/4-20ma/ModbusMaster/issues/96#issuecomment-368576230
vielleicht hilft das, DE & RE hab ich nicht genutzt - glaube ich...

Gruß

**Reply by Jaaxy** [reply]
Ja mit der Board Variante des RS485 habe ich das auch schon ein paarmal probiert. Beim ESP allerdings immer ohne Erfolg, da die Umschaltung der Sende-Empfangsrichtung im Tasmota nicht implementiert sein soll (laut einem Beitrag zum SDM530 MODBUS). Mit der abgebildeten Baugruppe oben, funktionierts mit dem ESP zumindest bei SDM530, dann sollte es für EPEver auch möglich sein. 

So viel ich weiß sogar ohne Levelshifting, aber ich schau mal morgen bei mir in der UV. 
Da habe ich den MODBUS mit dem SDM530 am laufen.

`+1,3,m,16,115200,,1,10,r010431100001,r0104311A0001,r0104331A0001,r0104331B0001,r010431000001,r010431010001,r010431020001,r0104310C0001,r0104310D0001,r0104310E0001,r010433040001,r0104330A0001,r0104330C0001,r010433120001,r010433030001
`
Dachte eigentlich das ich es mit dieser Anpassung hinbekomme, bis jetzt allerdings negativ... Versuche morgen nochmal Widerständen und Levelshifting was zu machen. Schreibe wenn´s klappt

**Comment by Jaaxy** [comment]
![image](https://user-images.githubusercontent.com/24737135/148597741-ddb50f69-ca62-4997-a255-f2e275901547.png)
So habe ich die Schaltung aufgebaut. Dem ESP8266 habe ich einen AMS verpasst um die Schaltung mit 5V beschalten zu können.
![image](https://user-images.githubusercontent.com/24737135/148598299-b3b5f733-7fa2-4fcb-979f-de1c7366584e.png)
Ein entsprechendes Gehäuse und wahrscheinlich auch ein PCB bereite ich gerade vor.

Den Code habe ich nochmal angepasst... 



Ich denke mal es lag daran das ich das RS485 Modul mit 3,3V betrieben hatte. Nachdem es nun auf 5V läuft, funktioniert die Übertragung auch.

**Reply by calasci** [reply]
Come faccio a caricare i script ?  Grazie

**Comment by Jaaxy** [comment]
Hier das neue Script:


```
>D
;
>B
smlj=0
;->sensor53 r
>R
smlj=0
;
>S
if upsecs>30
then
smlj=1
endif
;
>M 1
;
+1,1,m,16,115200,,3,10,r010431100001,r0104311A0001,r0104331A0001,r0104331B0001,r010431000001,r010431010001,r010431020001,r0104310C0001,r0104310D0001,r0104310E0001,r010433040001,r0104330A0001,r0104330C0001,r010433120001,r010433030001
;
1,=h Batterie
1,010402SSssxxxx@i1:100,Temp,°C,Batt Temperature,2
1,010402SSssxxxx@i2:1,SOC,%,Batt Humidity,0
1,010402SSssxxxx@i3:100,Spannung,V,Batt Voltage,2
1,010402SSssxxxx@i4:100,Strom,A,Batt Current,2
1,=h PV-Modul
1,010402SSssxxxx@i5:100,Spannung,V,PV Voltage,2
1,010402SSssxxxx@i6:100,Strom,A,PV Current,2
1,010402SSssxxxx@i7:100,Power,W,PV Power,2
1,=h LOAD
1,010402SSssxxxx@i8:100,Spannung,V,Load Voltage,2
1,010402SSssxxxx@i9:100,Strom,A,Load Current,2
1,010402SSssxxxx@i10:100,Power,W,Load Power,2
1,=h Energie
1,010402SSssxxxx@i11:100,Verbrauch_daily,KWh,Today out,2
1,010402SSssxxxx@i12:100,Verbrauch_total,KWh,Total out,2
1,010402SSssxxxx@i13:100,Ertrag_daily,KWh,Today in,2
1,010402SSssxxxx@i14:100,Ertrag_total,KWh,Total in,2
1,010402SSssxxxx@i15:100,max.Batt.,V,Batt max Voltage,2
;
#`
```

Hatte festgestellt, dass bei dem Orginal Script 3x Voltage im JSON/MQTT übertragen wurde. Dies habe ich jetzt noch angepasst. 

Außerdem habe ich eine Platine erstellt die ich mit 5V/12V wahlweise betreiben kann und in ein Hutschienengehäuse passt:
![epever-rs485](https://user-images.githubusercontent.com/24737135/165132647-9694d7fe-2ebc-41d8-9625-a749136cd0be.png)
.

**Reply by corbolais** [reply]
Hey @Jaaxy 

> Außerdem habe ich eine Platine erstellt die ich mit 5V/12V wahlweise betreiben kann und in ein Hutschienengehäuse passt: ![epever-rs485](https://user-images.githubusercontent.com/24737135/165132647-9694d7fe-2ebc-41d8-9625-a749136cd0be.png) .

Looks promising. Would you mind to share the schematic and layout files? I'd like to use and improve on it.

Cheers

**Reply by Jaaxy** [reply]
[Here is the Link to Layout](https://oshwlab.com/haderralf/epever-rs485-esp_version2_copy)

Please check ... hope it works

**Reply by salopette** [reply]
Ich habe jetzt alles geschafft zu konfigurieren, aber bei mir kommen keine Daten von Tracer.

Habe ein D1 Mini und diesen Adapter

https://user-images.githubusercontent.com/24737135/148094582-2683af58-f6ee-499f-9f10-94214fec58c7.png

und natürlich den Script von dir.
Muss ich noch etwas anpassen?

![sshot-3](https://user-images.githubusercontent.com/32480002/194626456-839eaab5-7485-4f03-83ed-42d025276103.jpg)

**Reply by Doerk32** [reply]
does ist now works? i have the same problem.

**Comment by Jaaxy** [comment]
[Here is the Link to the Layout](https://oshwlab.com/haderralf/epever-rs485-esp_version2_copy)

I was thinking of a larger version with relay outputs in a larger case. I think that´s the next step

**Comment by salopette** [comment]
> Hi guys,
> 
> my solution for data-logging via modbus, maybe someone need ... or add as an example @ [Smart Meter Descriptor examples](https://tasmota.github.io/docs/Smart-Meter-Interface/#smart-meter-descriptor-examples)
> 
> RS485 Adapter 485 Serieller Port UART Level Converter via Serial2 @ESP32
> 
> as scipting:
> 
> ```
> >D
> ;
> >B
> smlj=0
> ;->sensor53 r
> >R
> smlj=0
> ;
> >S
> if upsecs>30
> then
> smlj=1
> endif
> ;
> >M 1
> ;
> +1,16,m,16,9600,,17,10,r010431100001,r0104311A0001,r0104331A0001,r0104331B0001,r010431000001,r010431010001,r010431020001,r0104310C0001,r0104310D0001,r0104310E0001,r010433040001,r0104330A0001,r0104330C0001,r010433120001,r010433030001
> ;
> 1,=h Batterie
> 1,010402SSssxxxx@i1:100,Temp,°C,Temperature,2
> 1,010402SSssxxxx@i2:1,SOC,%,Humidity,0
> 1,010402SSssxxxx@i3:100,Spannung,V,Voltage,2
> 1,010402SSssxxxx@i4:100,Strom,A,Current,2
> 1,=h PV-Modul
> 1,010402SSssxxxx@i5:100,Spannung,V,Voltage,2
> 1,010402SSssxxxx@i6:100,Strom,A,PV Current,2
> 1,010402SSssxxxx@i7:100,Power,W,PV Power,2
> 1,=h LOAD
> 1,010402SSssxxxx@i8:100,Spannung,V,Voltage,2
> 1,010402SSssxxxx@i9:100,Strom,A,Current Load,2
> 1,010402SSssxxxx@i10:100,Power,W,Power,2
> 1,=h Energie
> 1,010402SSssxxxx@i11:100,Verbrauch_daily,KWh,Today,2
> 1,010402SSssxxxx@i12:100,Verbrauch_total,KWh,Total out,2
> 1,010402SSssxxxx@i13:100,Ertrag_daily,KWh,Today in,2
> 1,010402SSssxxxx@i14:100,Ertrag_total,KWh,Total,2
> 1,010402SSssxxxx@i15:100,max.Batt.,V,Voltage_0,2
> ;
> #`
> ```
> 
> ![Unbenannt](https://user-images.githubusercontent.com/57873076/127740947-43d17081-c079-45dd-b1c0-d64616290d6e.PNG)

Hallo, ich bin durch google auf das Thema gestosse! kann ich einfach Tasmota auf esp flashen TX und RX konfigurieren und dann kommen die werte von Epever? Oder muss noch etwas extra programiert werden?

Hab einen Tracer und würde gerne die Daten über tasmota auslesen.

Danke

**Reply by Jaaxy** [reply]
you have to activate scripting at the tasmota software. 
here is the usecase [https://tasmota.github.io/docs/Smart-Meter-Interface/](https://tasmota.github.io/docs/Smart-Meter-Interface/)

Scipting is not activated at the standard tasmota.bin, so you have to compile it new

**Comment by salopette** [comment]
and if I use tasmota sensor.bin, does that work?
Unfortunately I can't create a bin myself.

**Reply by barbudor** [reply]
Everybody can create a bin him/herself. There a even web based tools like TasmoCompiler

No regarding scripting there is an unofficial build tasmota-scriting
You can flash it using the web installer from tasmota.github.io/install
Choose the right one in the drop list

**Comment by salopette** [comment]
i think i need something like this.
but I can't find sdm120.

https://github.com/arendst/Tasmota/issues/2803

https://github.com/jarcikw/SDM120_Homeassistant/wiki/SDM120-&-nodemcu-&-RS485-wiring

**Reply by Jaaxy** [reply]
the sdm120 only activated if tasmota compiled with modbus sdm120.

The RS485 of SDM120 is configured to the rx/tx strings of the Eastron SDM120.  I think you need to use this script for EPEver, because the codes are not the same. The wiring can be used like the page you linked.

**Comment by salopette** [comment]
I think I set the voltage to 3.3v, maybe that's why there are no values?

**Comment by Jaaxy** [comment]
i have running 3 of this esp8266 to rs485 adapter. They linked to EPEver, SDM530 Eastron and EM24 Carlo Gavazzi. All RS485 running on 5V

**Comment by radiergimmy** [comment]
Where did you get the modbus register information from? I'm also interest in how to set/change the load relay state if that's possible

**Reply by corbolais** [reply]
Hi @radiergimmy 
Great! You managed to get what you needed. Thank you for sharing your script.

Hint: You could use triple backticks (\`) to format code pasted in. This prevented the indentation as used in a reply.

It would look like this then:

```
>D
SOCmin=50
SOC=0
lON="r01050002FF00"
lOFF="r010500020000"
lSwitch=0
lState=0
>B
smlj=0
->sensor53 r
>R
smlj=0
>S
if upsecs>30
then
smlj=1
endif
>W
bu(lSwitch "Load ON" "Load OFF")
nm(0 100 1 SOCmin "SOC (min)")
>T
SOC=EP#bSOC
lState=EP#lStatus

if ((chg[lState]>0) and (lSwitch!=lState))
then
lSwitch=lState
endif

if (((chg[SOC]>0) or (chg[SOCmin]>0)) and (SOC>0) and (SOC<SOCmin))
then
if lState>0
then
print EPEVER OFF - SOC %SOC% < %SOCmin%
sml(1 3 lOFF)
lSwitch=0
else
if lSwitch>0
then
lSwitch=0
endif
endif
endif

if chg[lSwitch]>0
then
if lSwitch>0
then
if ((SOC==0) or (SOC>=SOCmin))
then
print EPEVER ON - SOC %SOC%
sml(1 3 lON)
else
print EPEVER ON prevented - SOC %SOC% < %SOCmin%
lSwitch=0
endif
else
print EPEVER OFF - SOC %SOC%
sml(1 3 lOFF)
endif
endif

>M 1
+1,3,m,16,115200,EP,1,,r010432020001,r010432000001,r0104311A0001,r010431100001,r0104331B0001,r0104331A0001,r010433020001,r010433030001,r010431010001,r010431000001,r010431020001,r010433040001,r0104330A0001,r0104330C0001,r010433120001
1,=h<hr/>Last
1,010402SSssxxxx@i1:1,Status,,lStatus,0
1,=h<hr/>Batterie
1,010402SSssxxxx@i2:1,Status,,bStatus,0
1,010402SSssxxxx@i3:1,SOC,%%,bSOC,0
1,010402SSssxxxx@i4:100,Temp,°C,bTemperature,2
1,010402SSssxxxx@i5:100,Strom,A,bCurrent,2
1,010402SSssxxxx@i6:100,Spannung,V,bVoltage,2
1,010402SSssxxxx@i7:100,Spannung (max),V,bVoltageMax,2
1,010402SSssxxxx@i8:100,Spannung (min),V,bVoltageMin,2
1,=h<hr/>PV-Modul
1,010402SSssxxxx@i9:100,Strom,A,pCurrent,2
1,010402SSssxxxx@i10:100,Spannung,V,pVoltage,2
1,010402SSssxxxx@i11:100,Power,W,pPower,2
1,=h<hr/>Energie
1,010402SSssxxxx@i12:100,Verbrauch (Tag),kWh,dConsumption,2
1,010402SSssxxxx@i13:100,Verbrauch (Total),kWh,tConsumption,2
1,010402SSssxxxx@i14:100,Ertrag (Tag),kWh,dProd,2
1,010402SSssxxxx@i15:100,Ertrag (Total),kWh,tProd,2
#
```

**Reply by calasci** [reply]
salve , dove posso trovare il file per visualizzare  EPEVER  , non riesco a capire come caricare lo script 

hi , where can i find the file to view EPEVER , i can't figure out how to load the script

grazie

**Comment by corbolais** [comment]
@radiergimmy What type of EPever SolarTracer are you working with?

**Reply by radiergimmy** [reply]
I'm using a Tracer 5415AN.
Also reached out to EPEVER to get the latest version of their modbus register documentation

**Reply by corbolais** [reply]
> I'm using a Tracer 5415AN. Also reached out to EPEVER to get the latest version of their modbus register documentation

Great news. 

Did you receive an updated version of specs for this Tracer? I'm using a T4215BN not without hickups. See https://github.com/arendst/Tasmota/discussions/16422.

**Reply by radiergimmy** [reply]
Yes I did, but it's not much different from this one:
https://github.com/softwarecrash/EPEver2MQTT/blob/main/Protocol/MODBUS-Protocol-v25.pdf

**Reply by corbolais** [reply]
> Yes I did, but it's not much different from this one: https://github.com/softwarecrash/EPEver2MQTT/blob/main/Protocol/MODBUS-Protocol-v25.pdf

Great, didn't know this one. Thank you for sharing. What are the changes in between revisions?

**Reply by radiergimmy** [reply]
I've uploaded the 2.6 version to file.io
https://file.io/040Lik9kcmrj

**Reply by calasci** [reply]
please link ?
il link non funziona

**Comment by Kantulak** [comment]
Few notes:
-Epever Tracer (4210AN) supplies max 200mA on RS485 and the ESP module can consume more - especially in case of poor wifi connection - instead of 5V VCC from RS485, my suggestion is to power the module directly from the battery - connected to an appropriate step-down converter. (also some capacitance may help to avoid strange reboots - tested for ESP-01s)
- I am also using a bi-directional level converter (5V-3.3V) for TX/RX
- Full Tasmota integration in the Home Assistant requires tweak in the yaml (maybe there is a better workaround): 
homeassistant:
  customize:
     sensor.ftv_link_epever_bc:
      state_class: measurement
      unit_of_measurement: A
      device_class: current
      icon: mdi:alpha-a-circle-outline
      friendly_name: Battery Charging Current
.. etc. repeated for every sensor in your script (with different MQTT json name, than "Temperature", "Current", "Voltage", "Power" etc.)

- my 100% longterm working script:
>D
>B
->sensor53 r
>M 1
+1,3,m,1,115200,Epever,1,10,01043100,01043101,01043102,0104310C,0104310D,0104310E,01043105,01043110
1,010404UUuuxxxx@i0:100,PV Voltage,V,Voltage,2
1,010404UUuuxxxx@i1:100,PV Current,A,Current,2
1,010404UUuuxxxx@i2:100,PV Power,W,Power,2
1,010404UUuuxxxx@i3:100,Output Voltage,V,lv,2
1,010404UUuuxxxx@i4:100,Output Current,A,lc,2
1,010404UUuuxxxx@i5:100,Output Power,W,lp,2
1,010404UUuuxxxx@i6:100,Charging Current,A,bc,2
1,010404SSssxxxx@i7:100,Temperature,°C,Temperature,1
#

**Reply by corbolais** [reply]
Thanks for sharing.

Not using Tracer supply lines. Deployed known-good step-down with ESP-12f, ESP32.

I thought about adding a level converter for the UART, didn't do it yet, though. Will try. This might actually be a valid thing.

Not sure about the HA mod. I'm experiencing no issues for the values the ESP is actuall yreceiving. Display in HA is fine.

Will check your script with my setup. Main diff is you're requesting 4 byte answers in general.

Keep you posted.

**Reply by corbolais** [reply]
> I thought about adding a level converter for the UART, didn't do it yet, though. Will try. This might actually be a valid thing.

No difference. All the same, every second Tx gets no Rx at all. Very consistently.

**Comment by LR-Serie3** [comment]
Moin zusammen,
ich versuche seit Tagen zwei verschiedene Modbus-Stromzähler über Tasmota zum Laufen zu bringen. Leider vergebens. Ich habe einen Orno WE517 und einen Eastron SDM 72M-D, die jeweils über einen WEMOS D1 mini und einen RS485 auf TTL Adapter angeschlossen sind (Schaltplan genau wie oben im Beitrag von Jaaxy vom 07.01.22).
Der Wemos ist mit einer angepassten Tasmota Version geflasht die ich mit Gidpod gebaut habe. Dabei habe ich anfangs die häufig dokumentierte SML Variante ohne Erfolg versucht. Danach habe ich gelesen, dass beide Zähler auch direkt unterstützt werden sollten und habe die Zähler direkt in der entsprechenden Config Datei (my_user_config.h) aktiviert. -> siehe auch hier: https://www.michael-heck.net/index.php/smarthome/haussteuerung/sdm-630-mit-tasmota-auslesen

Stellt man dann in den Einstellungen den TX und RX GPIO ein, erscheint auch die Messwerttabelle im Tasmota. Allerdings sind alle Werte NULL.
Die TX und RX LEDs auf dem RS485 Board blinken regelmäßig (etwas jede Sekunde). Mit dem Oszi konnte ich auch sehen, dass Signale auf dem Wemos TX&RX Pin ankommen.
Die Daten werden dann per mqtt zum IObroker übertragen werden. Aber auch hier ist natürlich alles NULL.

Hat jemand einen Tipp für mich. Ich habe bereits das komplette Netz durchforstet aber keine Lösung gefunden.
Danke Dirk

**Reply by Noschvie** [reply]
Hallo Dirk
das hier ist ein englischsprachiges Forum. Poste dein Anliegen beispielsweise hier : https://forum.creationx.de/forum
Stimmen die Baudrate Parameter überein?

**Reply by Jaaxy** [reply]
> Moin zusammen, ich versuche seit Tagen zwei verschiedene Modbus-Stromzähler über Tasmota zum Laufen zu bringen. Leider vergebens. Ich habe einen Orno WE517 und einen Eastron SDM 72M-D, die jeweils über einen WEMOS D1 mini und einen RS485 auf TTL ...

Die Baudrate bei SDM630 ist in der Regel auf 9600, bei mir im Script ist 115000 eingestellt.

Außerdem, vertausch mal RX/TX in der Config. Vielleicht bringts die Lösung, ansonsten sind die Abschlusswiderstände evtl. drigend nötig. Kommt auf das Verbindungskabel und die Länge an. Ich habe 2Mbit Rangierdraht verwendet der ist nochmal abgeschirmt.

Gruß

**Comment by calasci** [comment]
> Hi guys,
> 
> my solution for data-logging via modbus, maybe someone need ... or add as an example @ [Smart Meter Descriptor examples](https://tasmota.github.io/docs/Smart-Meter-Interface/#smart-meter-descriptor-examples)
> 
> RS485 Adapter 485 Serieller Port UART Level Converter via Serial2 @ESP32
> 
> as scipting:
> 
> ```
> >D
> ;
> >B
> smlj=0
> ;->sensor53 r
> >R
> smlj=0
> ;
> >S
> if upsecs>30
> then
> smlj=1
> endif
> ;
> >M 1
> ;
> +1,16,m,16,9600,,17,10,r010431100001,r0104311A0001,r0104331A0001,r0104331B0001,r010431000001,r010431010001,r010431020001,r0104310C0001,r0104310D0001,r0104310E0001,r010433040001,r0104330A0001,r0104330C0001,r010433120001,r010433030001
> ;
> 1,=h Batterie
> 1,010402SSssxxxx@i1:100,Temp,°C,Temperature,2
> 1,010402SSssxxxx@i2:1,SOC,%,Humidity,0
> 1,010402SSssxxxx@i3:100,Spannung,V,Voltage,2
> 1,010402SSssxxxx@i4:100,Strom,A,Current,2
> 1,=h PV-Modul
> 1,010402SSssxxxx@i5:100,Spannung,V,Voltage,2
> 1,010402SSssxxxx@i6:100,Strom,A,PV Current,2
> 1,010402SSssxxxx@i7:100,Power,W,PV Power,2
> 1,=h LOAD
> 1,010402SSssxxxx@i8:100,Spannung,V,Voltage,2
> 1,010402SSssxxxx@i9:100,Strom,A,Current Load,2
> 1,010402SSssxxxx@i10:100,Power,W,Power,2
> 1,=h Energie
> 1,010402SSssxxxx@i11:100,Verbrauch_daily,KWh,Today,2
> 1,010402SSssxxxx@i12:100,Verbrauch_total,KWh,Total out,2
> 1,010402SSssxxxx@i13:100,Ertrag_daily,KWh,Today in,2
> 1,010402SSssxxxx@i14:100,Ertrag_total,KWh,Total,2
> 1,010402SSssxxxx@i15:100,max.Batt.,V,Voltage_0,2
> ;
> #`
> ```
> 
> ![Unbenannt](https://user-images.githubusercontent.com/57873076/127740947-43d17081-c079-45dd-b1c0-d64616290d6e.PNG)
dove metto lo script ?

**Reply by corbolais** [reply]
> dove metto lo script ?

See the IP address in you browser's address bar? Then go to `http://<YOUR_IP_HERE>/s10`
This only works if you have a Tasmota running with `#USE_SCRIPT` compiled in.
Remember to add the tick mark to `Script enable` above and hit `Save` below.

Cheers
